msr/msfr/thermochemistry/thermo.i

```diff
@@ -0,0 +1,2154 @@
+    combinatorial_geometry = 'y>1.3'
+    new_sideset_name = 'extraction'
+    included_boundaries = 'reflector_wall'
+    replace = true
+    input = 'restart'
+  []
+[]
+
+[GlobalParams]
+  elements = 'F Li U Th Ni Nd Ce La Cs Xe I Kr Ne'
+  output_phases = 'MSFL gas_ideal Ni_Solid_FCC(s) U_Solid-A(s)'
+  #output_phases = 'ALL'
+  output_species = 'MSFL:UF3 MSFL:U3+//I MSFL:U2F8 MSFL:U2//I MSFL:U[VII]//F MSFL:U[VII]//I MSFL:U[VI]//F MSFL:U[VI]//I  MSFL:NiF2 MSFL:Ni//I gas_ideal:CsI gas_ideal:CsF gas_ideal:I gas_ideal:I2 gas_ideal:I2Ni gas_ideal:ILi gas_ideal:Xe gas_ideal:Kr gas_ideal:Ne'
+  #output_species = 'ALL'
+  output_element_potentials = 'ALL'
+  output_vapor_pressures = 'vp:gas_ideal:CsI vp:gas_ideal:CsF vp:gas_ideal:I vp:gas_ideal:I2 vp:gas_ideal:I2Ni vp:gas_ideal:ILi'
+  #output_vapor_pressures = 'ALL'
+[]
+
+[ChemicalComposition]
+  thermofile = MSTDTC_Noble_metal_gases.dat
+  tunit = K
+  punit = Pa
+  munit = moles
+  temperature = tfuel_nod
+  pressure = pressure_nod
+  reinitialization_type = nodal
+[]
+
+[AuxVariables]
+  [tfuel]
+    order = CONSTANT
+    family = MONOMIAL
+    #initial_condition = 936.00
+    initial_from_file_var = T_fluid
+  []
+  [tfuel_nod]
+    order = FIRST
+    family = Lagrange
+  []
+  [pressure]
+    order = CONSTANT
+    family = MONOMIAL
+    #initial_condition = 1.3042
+    initial_from_file_var = pressure
+  []
+  [pressure_nod]
+    order = FIRST
+    family = Lagrange
+  []
+[]
+
+[AuxKernels]
+  [ProjAuxT]
+    type = ProjectionAux
+    v = tfuel
+    variable = tfuel_nod 
+    execute_on = INITIAL
+  []
+  [ProjAuxP]
+    type = ProjectionAux
+    v = pressure
+    variable = pressure_nod 
+    execute_on = INITIAL
+  []
+[] 
+
+[ICs]
+  [F]
+    type = FunctionIC
+    variable = F
+    function = FSumSet
+  []
+  [Li]
+    type = FunctionIC
+    variable = Li
+    function = LiSumSet
+  []
+  [U]
+    type = FunctionIC
+    variable = U
+    function = USumSet
+  []
+  [Th]
+    type = FunctionIC
+    variable = Th
+    function = ThSumSet
+  []  
+  [Ni]
+    type = ConstantIC
+    variable = Ni
+    #value = 7.1e-03
+    value = 1
+  []
+  [Xe]
+    type = FunctionIC
+    variable = Xe
+    function = XeSumSet
+  []
+  [Nd]
+    type = FunctionIC
+    variable = Nd
+    function = NdSumSet
+  []  
+  [Ce]
+    type = FunctionIC
+    variable = Ce
+    function = CeSumSet
+  []  
+  [La]
+    type = FunctionIC
+    variable = La
+    function = LaSumSet
+  []  
+  [Cs]
+    type = FunctionIC
+    variable = Cs
+    function = CsSumSet
+  []  
+  [I]
+    type = FunctionIC
+    variable = I
+    function = ISumSet
+  []  
+  [Kr]
+    type = FunctionIC
+    variable = Kr
+    function = KrSumSet
+  []  
+  [Ne]
+    type = FunctionIC
+    variable = Ne
+    function = NeSumSet
+  []  
+[]
+
+[Problem]
+  solve = false
+  allow_initial_conditions_with_restart = true
+[]
+
+[Executioner]
+  #type = Transient
+  type = Steady
+[]
+
+[Outputs]
+  csv = true
+  exodus = true
+[]
+
```

Comment:
I would add sections like
```
*******************************************
* Postprocessing the ...
*******************************************
```
because it s tough to navigate a 2000 lines input

Reply:
Included in new push